core/templates/pages/exploration-editor-page/services/exploration-rights.service.ts

```diff
@@ -230,6 +230,20 @@ export class ExplorationRightsService {
   }
 
   removeRoleAsync(memberUsername: string): Promise<void> {
+    const initialState = {
```

Comment:
Can you do this in a way that doesn't assume there will always be three categories of usernames? For example, store just the name of the field containing the username in question and remove the member username from it, then add the member username back later to the same category if needed (or reinstate the old version of the category). 

Other fields should not be affected here.

Reply:
Done

Reply:
Not done. I don't expect to see reference to three things here. Please make the code more general so that if we add a new username category, we don't have to update this code. (As a bonus, this will also reduce the repeated logic.)